openshift_metrics/invoice.py

```diff
@@ -0,0 +1,203 @@
+import math
+from dataclasses import dataclass, field
+from collections import namedtuple
+from typing import List
+from decimal import Decimal, ROUND_HALF_UP
+
+# GPU types
+GPU_A100 = "NVIDIA-A100-40GB"
+GPU_A100_SXM4 = "NVIDIA-A100-SXM4-40GB"
+GPU_V100 = "Tesla-V100-PCIE-32GB"
+GPU_UNKNOWN_TYPE = "GPU_UNKNOWN_TYPE"
+
+# GPU Resource - MIG Geometries
+# A100 Strategies
+MIG_1G_5GB = "nvidia.com/mig-1g.5gb"
+MIG_2G_10GB = "nvidia.com/mig-2g.10gb"
+MIG_3G_20GB = "nvidia.com/mig-3g.20gb"
+WHOLE_GPU = "nvidia.com/gpu"
+
+# SU Types
+SU_CPU = "OpenShift CPU"
+SU_A100_GPU = "OpenShift GPUA100"
+SU_A100_SXM4_GPU = "OpenShift GPUA100SXM4"
+SU_V100_GPU = "OpenShift GPUV100"
+SU_UNKNOWN_GPU = "OpenShift Unknown GPU"
+SU_UNKNOWN_MIG_GPU = "OpenShift Unknown MIG GPU"
+SU_UNKNOWN = "Openshift Unknown"
+
+ServiceUnit = namedtuple("ServiceUnit", ["su_type", "su_count", "determinig_resource"])
+
+
+class Pod:
+    """Object that represents a pod"""
+
+    def __init__(
+        self,
+        start_time: int,
+        duration: int,
+        cpu_request: Decimal,
+        gpu_request: Decimal,
+        memory_request: Decimal,
+        gpu_type: str,
+        gpu_resource: str,
+    ):
+        self.start_time = start_time
+        self.end_time = start_time + duration
```

Comment:
@naved001 I'd like to ask about `end_time`. Are there other uses of `end_time` other than to calculate `duration`? Why doesn't `get_runtime()` return `duration/3600` directly?

Reply:
@knikolla thanks, that's a good idea. 

@QuanMPhm I am basically laying the groundwork so that I can ignore certain hours to not bill for during start_time and end_time